src/main/java/duke/parser/ParserManager.java

```diff
@@ -0,0 +1,81 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.IncompleteCommandError;
+import duke.exception.UnknownCommandException;
+import duke.task.TaskList;
+
+import java.util.Optional;
+
+public class ParserManager {
+    private TaskList taskList;
+
+    /**
+     * Parse given user input and returns Optional containing null or valid command
+     * @param taskList - list containing all existing tasks
+     * @param fullCommand - command given by user input
+     * @return Optional containing either valid command or null (when exception is thrown)
+     */
+    public Optional<Command> parseCommand(TaskList taskList, String fullCommand) {
+        this.taskList = taskList;
+        String[] commandDescription = fullCommand.trim().split("\\s+", 2);
+        commandDescription[0] = commandDescription[0].toLowerCase();
+        String taskName = commandDescription[0];
+        try {
+            // Special case since only word "list" is required
+            if(taskName.equals("list")) { return ListCommandParser.parse(fullCommand); }
+            // Checks if command is empty
+            this.checkCommandEmpty(commandDescription);
+
+            switch (taskName) {
```

Comment:
Since commands are limited to a fixed set of values, maybe it would be a good idea to use an enum?

I think it also offers a trivial runtime benefit as you only call CommandsEnum.valueOf(taskName) once ( and then catch IllegalArgumentException if invalid), and compare for enum equality. 
( In C / C++ / C# enums are fundamentally implemented with integers, _I believe_ it is the same for Java but I cant find a good source on it thus far )

(instead of comparing string equality till the correct string is reached)